### DISCUSSION TITLE
How to remove underline on H1 text style in readme file

### DISCUSSION THREAD

**Discussion by bravo-phantom** [community/community#132530] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

**Question:**
- How to remove H1 text style's underline?
- Please share a couple of other ways to format text as well; like h1, h2, h3, etc.

**My Code**
`<h1> My Text </h1>`

**My Repository Which Has Problem In ReadMe**
[Phone Live Wallpapers](https://github.com/bravo-phantom/ps_PhoneLiveWallpapers)

**Important - Please Read**
I know nothing about HTML or CSS. I'm a Unity Dev and also an expert in C# that's why I understand some of this programming. But please answer in simple, plain and detailed words about what and where to type. Thanks :)

**Comment by devmuse7** [comment]


**Reply by bravo-phantom** [reply]
> In a md file(README.md), text withing angle brackets h1 tag can sometimes be interpreted as HTML which lead to unexpected formatting. Md file itself doesn't directly support styling like CSS.
> 
> **To fix your issue:**
> 
> **Do this:** `# Phone Live Wallpapers`
> 
> **Instead of this:** `<h1 Phone Live Wallpapers h1> `
> 
> -- Please mark it as answered below, if you find this useful.📌

Thanks for your reply. But it didn't fix the issue. It looks the same as when we were using h1, but the line is still there. Maybe if there is a way to add a smaller line that doesn't overlap the images or just use another style.

**Reply by bravo-phantom** [reply]
> You can `**Phone Live Wallpapers**` That underline is coming because you're considering this text as Heading

This works to bold the text. But I want to increase the size; bold is not the objective. Can you please share how to increase size of the text.

**Comment by davevad93** [comment]
Hi @bravo-phantom ,

In GitHub's Markdown a line is given by default for `h1` and `h2` tags (or # and ##) and there's neither a way to apply `CSS` styles to increase text size, because style attributes are totally ignored in GitHub `md` files, for example, changing the text color wouldn't work either. If you want a heading with no line the only solution would be using a `h3` tag or ### but obviously it would look smaller.

Here's an example of how would it look in your IDE a `<h3>` with the `font-size` property applied:

![0](https://github.com/user-attachments/assets/f3dd1629-f868-41e0-9c5f-7c5a67c845d4)
![1](https://github.com/user-attachments/assets/1720fb50-8a8f-413d-93aa-4e5f93abb476)

This is how the same example looks like in GitHub:

![2](https://github.com/user-attachments/assets/0e9579c0-2b4d-4ed8-9f89-9ceb672cea97)
![3](https://github.com/user-attachments/assets/7b1c7014-55a4-4654-93ef-fda7381b4f5f)

As you can see, in GitHub the `font-size` property isn't recognized.

**Reply by bravo-phantom** [reply]
> Hi @bravo-phantom ,
> 
> In GitHub's Markdown a line is given by default for `h1` and `h2` tags (or # and ##) and there's neither a way to apply `CSS` styles to increase text size, because style attributes are totally ignored in GitHub `md` files, for example, changing the text color wouldn't work either. If you want a heading with no line the only solution would be using a `h3` tag or ### but obviously it would look smaller.
> 
> Here's an example of how would it look in your IDE a `<h3>` with the `font-size` property applied:
> 
> ![0](https://private-user-images.githubusercontent.com/94308234/349720101-f3dd1629-f868-41e0-9c5f-7c5a67c845d4.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNjM4NTAsIm5iZiI6MTcyMTI2MzU1MCwicGF0aCI6Ii85NDMwODIzNC8zNDk3MjAxMDEtZjNkZDE2MjktZjg2OC00MWUwLTljNWYtN2M1YTY3Yzg0NWQ0LkpQRz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE4VDAwNDU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2ZjQ2ZjJiZGU4M2NhYTYzODAxMjE2YmI0Yjg1MTg4OGE2ZmQxMmVhNTkwYzI5ODY5MGYzZDFiMTE0MTljOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1cqahm8DUFonMDOaGUg-_qjG2JHNdl8yNagm-A14ncM) ![1](https://private-user-images.githubusercontent.com/94308234/349720328-1720fb50-8a8f-413d-93aa-4e5f93abb476.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNjM4NTAsIm5iZiI6MTcyMTI2MzU1MCwicGF0aCI6Ii85NDMwODIzNC8zNDk3MjAzMjgtMTcyMGZiNTAtOGE4Zi00MTNkLTkzYWEtNGU1ZjkzYWJiNDc2LkpQRz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE4VDAwNDU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkZGU2NWZiNjNjZjQyYjM5MWJiYmI3ZmJiOWZiZjAzZmZjMTc3NjhlYTRkZGM1NDZkZDkwZDM4MzA3MjQ4MDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uhhpJhyVg042JLQwK15enEXRk29zLMdaoYFVhkllbbQ)
> 
> This is how the same example looks like in GitHub:
> 
> ![2](https://private-user-images.githubusercontent.com/94308234/349720591-0e9579c0-2b4d-4ed8-9f89-9ceb672cea97.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNjM4NTAsIm5iZiI6MTcyMTI2MzU1MCwicGF0aCI6Ii85NDMwODIzNC8zNDk3MjA1OTEtMGU5NTc5YzAtMmI0ZC00ZWQ4LTlmODktOWNlYjY3MmNlYTk3LkpQRz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE4VDAwNDU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzZTk1ZjBjZGQ4Nzg3YTQ1ZjkxNmMxZTc5MGRlZjhiODI2MGRlNzZlYWFiNjYwMWI0YzE4ZmIwYzI1MDE1ZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.e0qc0XmtC7vCbD62MWbwiXYDFdpOwRw7Mi_mouRiqK0) ![3](https://private-user-images.githubusercontent.com/94308234/349720598-7b1c7014-55a4-4654-93ef-fda7381b4f5f.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNjM4NTAsIm5iZiI6MTcyMTI2MzU1MCwicGF0aCI6Ii85NDMwODIzNC8zNDk3MjA1OTgtN2IxYzcwMTQtNTVhNC00NjU0LTkzZWYtZmRhNzM4MWI0ZjVmLkpQRz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE4VDAwNDU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkZDQyN2Y0MjM0ZTk4MjY3YTEzMTVhODMwOTk2YmQ3MzFkZjcwMjZlYzBmNWEwOTliOGVhNzhmZTI5ODgzYWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qdiVwRHpxsj_oY9VKIG6gBlTaqL5ui6hTrri8t7Tep0)
> 
> As you can see, in GitHub the `font-size` property isn't recognized.

Thanks for the precise and to-the-point answer. Exactly what I was looking for :)

So in the end, I made an image of the text I wanted to type; saved it in the project folder; and used img=scr to type the file path and use the height and width to set the size.

**Reply by davevad93** [reply]
You're welcome @bravo-phantom , well I think your solution is good and looks great! 👍

**Reply by CodeByAidan** [reply]
@davevad93 See my solution, there's been a way.

**Comment by CodeByAidan** [comment, marked as answer]
I finally found a fix for this.... 4 hours of pain... I wrote an entire essay to help someone on Stack Overflow, this should explain everything.

https://stackoverflow.com/a/78845294/21147004

Also see here for the working example: https://gist.github.com/CodeByAidan/bb43bdb1c07c0933d8b67c23515fb912

So, if all you wanted was a `<h1> My Text </h1>`, it can be transformed into this:

```html
<div id="toc">
  <ul style="list-style: none">
    <summary>
      <h1> My Text </h1>
    </summary>
  </ul>
</div>
```

Result:

<div id="toc">
  <ul style="list-style: none">
    <summary>
      <h1> My Text </h1>
    </summary>
  </ul>
</div>

**Reply by bravo-phantom** [reply]
Nice :) Thanks for that, it would be handy in making future ReadMe for my repositories.

**Reply by davevad93** [reply]
Wow @CodeByAidan , that's awesome kudos to you and thanks for the tip!

**Reply by taqi110913** [reply]
Wow, it works, @CodeByAidan! Thank you!

**Reply by kshen0414** [reply]
@CodeByAidan chatgpt told me there was no way....one google search, this github issue was the first result, and your answer was exactly what the community needed...god bless you

**Comment by IndianPhotoAcademy** [comment]
If your repository is private, nobody else will be able to access it unless they are specifically invited. Check repository permissions, make sure you're pushing to the correct remote URL, and confirm your authentication (SSH or PAT) to make sure your push goes through. Use git remote -v to verify the repository destination.

**Comment by NaeemAbdullahAkram** [comment]
✅ Your Main Question: How to remove the underline from <h1> text?
In HTML, headings like <h1> don’t normally have underlines unless your README is being rendered with styles (like GitHub Markdown) or something is overriding the default styles.

But if your <h1> text is underlined, we can fix it using inline CSS — this means we will style it directly in the tag, like this:

html
Copy
Edit
<h1 style="text-decoration: none;">My Text</h1>

**Comment by finlayconnor27** [comment]
In README/Markdown, H1 itself isn’t underlined by default—if you’re seeing an underline, it’s coming from the theme/CSS, so you’d need to override it with `text-decoration: none` or adjust the renderer styling. It’s similar to structured workflows like [property turnover management services](https://realestatepropertymanagement.co.uk/services/real-estate-property-turnover/) where the output formatting is controlled by the system layer rather than the raw content.
